flagsmith-core.ts

```diff
@@ -788,6 +803,15 @@ const Flagsmith = class {
         if (method && method !== 'GET')
             options.headers['Content-Type'] = 'application/json; charset=utf-8';
 
+
+        if (this.appName) {
+            options.headers['X-Customer-Application-Name'] = this.appName;
```

Comment:
I would suggest mentioning Flagsmith somehow in this header, to describe what it's for.

Something like `X-Flagsmith-Application-Name` and `X-Flagsmith-Application-Version`. That way if customers have to do anything with this header, it's immediately obvious what it's for. It's obvious what the header is for if we're only ever calling APIs on `flagsmith.com`, but might not be as obvious for self-hosting customers.

Reply:
Also `X-` prefix for custom headers has been deprecated according to this [RFC](https://datatracker.ietf.org/doc/html/rfc6648)